File: Swift/add-strings.swift
class Solution {
    func addStrings(_ num1: String, _ num2: String) -> String {
        var chars1 = [Character](num1)
        var chars2 = [Character](num2)

        while chars1.count != chars2.count {
            if chars1.count > chars2.count {
                chars2.insert("0", at: 0)
            } else {
                chars1.insert("0", at: 0)
            }
        }

        let zeroValue = "0".unicodeScalars.first!.value
        var carry: UInt32 = 0
        var resChars = [Character]()

        for i in (0..<chars1.count).reversed() {
            let num1 = chars1[i].unicodeScalars.first!.value - zeroValue
            let num2 = chars2[i].unicodeScalars.first!.value - zeroValue
            var x = num1 + num2
            x += carry
            let y = x % 10
            carry = x / 10
            resChars.insert(Character(UnicodeScalar(y + zeroValue)!), at: 0)
        }

        return String(resChars)
    }
}

File: Python/add-strings.py
class Solution:
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        chars1 = list(num1)
        chars2 = list(num2)
        chars1.reverse()
        chars2.reverse()
        i, j, sumValue, carry = 0, 0, 0, 0
        res = list()

        while i < len(chars1) or j < len(chars2) or carry != 0:
            sumValue = carry

            if i < len(chars1):
                sumValue += int(chars1[i])
                i += 1
            if j < len(chars2):
                sumValue += int(chars2[j])
                j += 1

            carry = sumValue // 10
            sumValue = sumValue % 10

            res.append(str(sumValue))
        
        res.reverse()
        return ''.join(res)
